src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -115,11 +115,14 @@ public void addStudent(Student s) {
     public void setStudent(Student target, Student editedStudent) {
         requireNonNull(editedStudent);
 
+        // ID has changed, update IDs in assessments
         if (!target.getId().equals(editedStudent.getId())) {
             assessments.replaceStudent(target, editedStudent);
-            groups.replaceStudent(target, editedStudent);
         }
 
+        // Update group memberships
+        groups.replaceStudent(target, editedStudent);
```

Comment:
`replaceStudent` here removes all the student from all existing groups, then adds the student into the new groups. This should be done even when the ID did not change. If not, changes in group memberships are never updated.